### DISCUSSION TITLE
Why is `className` named as `UNSAFE_className` when using Combobox?

### DISCUSSION THREAD

**Discussion by deadcoder0904** [adobe/react-spectrum#2781] [Q&A] [answered]
I use Tailwind CSS & like it to quickly style it. I am also using combobox for a select because I like the autocomplete feature.

I wanted to style my existing Tailwind component to use Combobox with just style replaced however I saw it uses `UNSAFE_`

Now I wanna ask why is there `UNSAFE_`? Isn't it better to completely let me style it using Tailwind as it takes care of a11y as well & has better contrast ratios for readability.

I also can't get `className` prop on `Item` either so don't know how to style it? Do I need to use individual props there?

Ideally, I would love to use Tailwind with React Spectrum & React Aria.

**Comment by reidbarber** [comment, marked as answer]
In short, from the [Escape Hatch](https://react-spectrum.adobe.com/react-spectrum/styling.html#escape-hatches) section of the Styling page in the docs:

> The reasoning behind this is that future updates to Spectrum design may cause unintended breaking changes in products.

Also, React Spectrum components are intended to follow Adobe's Spectrum design system, and allowing loose extraneous styling can make it easy to create anti-patterns or violate the design system.

If you're building a custom ComboBox, you may want to look at using React Aria to build your own, so you can easily style it however you like. The useComboBox page has a [Styled examples](https://react-spectrum.adobe.com/react-aria/useComboBox.html#styled-examples) section that includes a [Tailwind CSS example](https://codesandbox.io/s/hardcore-moon-xzc4r?file=/src/ComboBox.tsx) that might be useful.

**Reply by snowystinger** [reply]
We do also use css variables for most properties with values such as color or size, you can use those to change the styling of your component if the Spectrum ComboBox is pretty close to what you need. You can look up those values through inspection. If you're on an Adobe product though, I do discourage this so that your ComboBox matches the rest of Adobe

**Reply by deadcoder0904** [reply]
makes sense. so react-aria is similar to headless ui & react-spectrum is adobe's design system. i do think react-spectrum docs could use a little improvement on the headline side.
